File: server/src/main/java/snaykuu.userInterface/ReplayPanel.java
package snaykuu.userInterface;

import snaykuu.gameLogic.RecordedGame;

import javax.swing.*;
import javax.swing.filechooser.FileNameExtensionFilter;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.io.File;

class ReplayPanel extends JPanel
{
	private SettingsWindow settingsWindow;
	private JButton loadAndPlay = new JButton("Load an old game and play it!");

	public ReplayPanel(SettingsWindow settingsWindow)
	{
		this.settingsWindow = settingsWindow;
		loadAndPlay.addActionListener(new ReplayListener());
		add(loadAndPlay);
	}

	private void startReplay(RecordedGame recordedGame)
	{
		try
		{
			new ReplayWindow(settingsWindow, recordedGame);
		}
		catch (NumberFormatException e)
		{
			JOptionPane.showMessageDialog(getParent(), "You must enter a valid amount of pixels per square");
		}
	}

	private class ReplayListener implements ActionListener
	{
		public void actionPerformed(ActionEvent event)
		{
			JFileChooser fileChooser = new JFileChooser("./replays");

			FileNameExtensionFilter filter = new FileNameExtensionFilter("Snaykuu Replay (.srp)", "srp");
			fileChooser.setFileFilter(filter);

			int returnValue = fileChooser.showOpenDialog(getParent());
			if (returnValue != JFileChooser.APPROVE_OPTION)
				return;

			File file = fileChooser.getSelectedFile();
			try
			{
				RecordedGame recordedGame = RecordedGame.Companion.load(file);
				startReplay(recordedGame);
			}
			catch (Exception e)
			{
				e.printStackTrace();
				JOptionPane.showMessageDialog(getParent(), e);
			}
		}
	}
}

File: server/src/main/kotlin/snaykuu/userInterface/ReplayPanel.kt
package snaykuu.userInterface

import snaykuu.gameLogic.RecordedGame
import snaykuu.gameLogic.RecordedGame.Companion.load
import java.awt.event.ActionEvent
import java.awt.event.ActionListener
import javax.swing.JButton
import javax.swing.JFileChooser
import javax.swing.JOptionPane
import javax.swing.JPanel

class ReplayPanel(
    private val settingsWindow: SettingsWindow
): JPanel() {

    init {
        val loadAndPlay = JButton("Load an old game and play it!")
        loadAndPlay.addActionListener(ReplayListener())
        add(loadAndPlay)
    }

    private fun startReplay(recordedGame: RecordedGame) {
        try {
            ReplayWindow(settingsWindow, recordedGame)
        } catch(e: NumberFormatException) {
            JOptionPane.showMessageDialog(parent, "You must enter a valid amount of pixels per square")
        }
    }

    private inner class ReplayListener: ActionListener {
        override fun actionPerformed(event: ActionEvent) {
            val fileChooser = JFileChooser("./replays")
            fileChooser.fileFilter = RecordedGame.fileNameFilter()

            val returnValue = fileChooser.showOpenDialog(parent)
            if(returnValue != JFileChooser.APPROVE_OPTION) return

            val file = fileChooser.selectedFile
            try {
                val recordedGame = load(file)
                startReplay(recordedGame)
            } catch(e: Exception) {
                e.printStackTrace()
                JOptionPane.showMessageDialog(parent, e)
            }
        }
    }
}